function-app/adb-to-purview/src/Function.Domain/Services/BlobClientFactory.cs

```diff
@@ -29,14 +34,57 @@ public BlobClientFactory(IConfiguration config, ILogger<BlobClientFactory> logge
             _blobServiceClient = new(
                 new Uri($"https://{accountName}.blob.core.windows.net"),
                 new DefaultAzureCredential(credentialConfig));
-            _blobContainerClient = _blobServiceClient.GetBlobContainerClient("ol-messages");
-            _blobContainerClient.CreateIfNotExists(Azure.Storage.Blobs.Models.PublicAccessType.None);
+            //_blobContainerClient = _blobServiceClient.GetBlobContainerClient("ol-messages");
+            //_blobContainerClient.CreateIfNotExists(Azure.Storage.Blobs.Models.PublicAccessType.None);
             logger.LogInformation("BlobClientFactory created for {accountName}", accountName);
         }
 
-        public BlobClient Create(string name)
+        public BlobClient Create(string containerName, string name)
         {
+            GetBlobContainer(containerName);
             return _blobContainerClient.GetBlobClient(name);
         }
+
+        public async Task UploadAsync(string containerName, string blobName, string jsonObject)
```

Comment:
@kantmani4 

This intent of this class was only for creation of BlobClient. I would move the new operations into a Scoped class BlobProvider or BlobBroker and move UPloadAsync,GetBlobsByHierarchiAsyncm DownloadBlobAsync.

Your new class would be added as scoped into DI and would take in the constructor the blobFactory

Reply:
In the service layer (OLConsoldiateEnrichSynapse) you would pass the scoped IBlobBroker via the constructor

Reply:
Done